src/test/java/com/puppycrawl/tools/checkstyle/bdd/TestInputConfiguration.java

```diff
@@ -104,9 +110,13 @@ public DefaultConfiguration createConfiguration() {
 
     public DefaultConfiguration createConfigurationWithoutFilters() {
         final DefaultConfiguration root = new DefaultConfiguration(ROOT_MODULE_NAME);
-        final DefaultConfiguration treeWalker =
-                new DefaultConfiguration(TreeWalker.class.getName());
         root.addProperty("charset", StandardCharsets.UTF_8.name());
+        DefaultConfiguration treeWalkerModule = new DefaultConfiguration(
+                TreeWalker.class.getName());
+        if (treeWalkerModuleInputConfiguration != null) {
+            treeWalkerModule = treeWalkerModuleInputConfiguration.createConfiguration();
+        }
+        final DefaultConfiguration treeWalker = treeWalkerModule;
```

Comment:
Too much variables about the same.
Use if-else assign directly to `DefaultConfiguration treeWalker`

Is it possible to avoid treeWalkerModuleInputConfiguration field ?
By treatment of Treewalker as just another child module.
And during construction of config search of it in children by name?

Reply:
I think removing treeWalkerModuleInputConfiguration would not be good, cause than we have to traverse in childModules list each time and if found we have to remove that from childModules also. So new Field should be good, there is just a  one simple check.

In short with this field time Complexity will reduce to just O(1) from O(n), n is number of child in list.

Reply:
In tests we do not care much of performance especially if that makes code easier.

Reply:
> In tests we do not care much of performance especially if that makes code easier. We value maintainability more than performance in this case.

Oh! That's why test are taking too much time to run

Reply:
No, because we parse files for each test. Parsing is most time consuming part.

Reply:
@romani, mvn clean verify taking too much time, is there any other command that will just check the updated files ?

Reply:
We are not that advanced, sorry.
How much it takes on your local?

You need to run verify only at the end, before push. In all other cases just run single test from your IDE

Reply:
it takes around 8-10 minutes,

![image](https://user-images.githubusercontent.com/61725413/201532956-e162344c-f594-45d2-8c05-4d5dc06bb437.png)

The main problem is to wait 8-10 minutes,  just to check errors of checkstyle

Reply:
BTW this is done

Reply:
It is not just checkstyle errors, there a lot of tool and validations.
CI do for you many hours of extra validations.

All together makes it a bit slow but no regression leaked to user - our paramount goal